### PR TITLE
Add extra fields to query statistics in listener

### PR DESCRIPTION
**Pull request by michaelcullum**

### CODE REVIEW

File: presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryStatistics.java [hunk 1]
```diff
@@ -155,12 +167,12 @@ public long getOutputRows()
...
     public long getWrittenBytes()
     {
-        return writtenBytes;
+        return logicalWrittenBytes;
     }
 
     public long getWrittenRows()
     {
-        return writtenRows;
+        return logicalWrittenRows;
```
Comment (findepi) [inline presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryStatistics.java]: if you didn't rename the getter, why did you rename the field?
or, the other way around, why don't you rename the getter too?

File: presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryStatistics.java [hunk 2]
```diff
@@ -62,14 +66,18 @@ public QueryStatistics(
             long totalRows,
             long outputBytes,
             long outputRows,
-            long writtenBytes,
-            long writtenRows,
+            long logicalWrittenBytes,
+            long logicalWrittenRows,
```
Comment (findepi) [inline presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryStatistics.java]: `writtenRows` was OK

File: presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryStatistics.java [hunk 3]
```diff
@@ -62,14 +66,18 @@ public QueryStatistics(
...
             List<StageGcStatistics> stageGcStatistics,
             int completedSplits,
             boolean complete,
             List<StageCpuDistribution> cpuTimeDistribution,
-            List<String> operatorSummaries)
+            List<String> operatorSummaries,
+            Duration elapsedTime,
+            int totalDrivers,
+            int totalTasks,
+            long physicalWrittenBytes)
```
Comment (findepi) [inline presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryStatistics.java]: put it next to "logical.."?

File: presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryStatistics.java [hunk 4]
```diff
@@ -83,14 +91,18 @@ public QueryStatistics(
...
-        this.writtenRows = writtenRows;
+        this.logicalWrittenBytes = logicalWrittenBytes;
+        this.logicalWrittenRows = logicalWrittenRows;
         this.cumulativeMemory = cumulativeMemory;
         this.stageGcStatistics = requireNonNull(stageGcStatistics, "stageGcStatistics is null");
         this.completedSplits = completedSplits;
         this.complete = complete;
         this.cpuTimeDistribution = requireNonNull(cpuTimeDistribution, "cpuTimeDistribution is null");
         this.operatorSummaries = requireNonNull(operatorSummaries, "operatorSummaries is null");
+        this.elapsedTime = elapsedTime;
```
Comment (findepi) [inline presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryStatistics.java]: `requireNonNull`

File: presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryStatistics.java [hunk 5]
```diff
@@ -62,14 +66,18 @@ public QueryStatistics(
             long totalRows,
             long outputBytes,
             long outputRows,
-            long writtenBytes,
-            long writtenRows,
+            long logicalWrittenBytes,
```
Comment (findepi) [inline presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryStatistics.java]: Document (comment) what's "logical bytes" and how it differs from "physical".
For example:
- i guess "logical" is before any compression. Am i right?
- i guess "logical" is before "unrolling" RLE or Dictionary blocks. Am i right?

File: presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryStatistics.java (hunk 1 above)
Comment (michaelcullum) [reply presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryStatistics.java]: I didn't want to introduce a BC break but thought this made things clearer within the class

File: presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryStatistics.java (hunk 5 above)
Comment (michaelcullum) [reply presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryStatistics.java]: Not sure it belongs here, would be better off in QueryStats.

```
    @JsonProperty
    public DataSize getLogicalWrittenDataSize()
    {
        return succinctBytes(
                operatorSummaries.stream()
                        .filter(stats -> stats.getOperatorType().equals(TableWriterOperator.class.getSimpleName()))
                        .mapToLong(stats -> stats.getInputDataSize().toBytes())
                        .sum());
    }
```

**Comment by michaelcullum** [comment]
Tests are failing but I can't see on Travis where the failure is?
